src/utils/validate-date-string.ts

```diff
@@ -0,0 +1,23 @@
+import { Temporal } from '@js-temporal/polyfill'
+import type { SupportedCalendar } from '../types'
+import { extractDatePartsFromDateString } from './extract-date-parts-from-date-string'
+
+export function validateDateString(
+    dateString: string,
+    { calendar = 'gregory' }: { calendar: SupportedCalendar } = {}
+): boolean {
+    try {
+        // will throw if the format of the date is incorrect
+        const { year, month, day } = extractDatePartsFromDateString(dateString)
+
+        // will throw if the year, month or day is out of range
+        Temporal.PlainDate.from(
```

Comment:
that looks good to me - there are two things that we'd need to change probably, but we can merge this first, integrate with calendar then improve:

1. I am not sure `Temporal.PlainDate.from` will work with custom calendars (nepali in our case) - for these, we'd need to map the string "nepali" to the actual calendar (similar to what happens in other places, and what's suggested in Temporal docs)
2. Instead of returning the actual error message, I suggest we return an error code from here (and console.error the message). The error code can then be used in the client, and also translated (since the library here is not tied to transifex or any localisation beyond what Intl.format natively provides)

Reply:
I added the missing implementation for 1. The other one we can do later, as you've suggested